### PR TITLE
Fix settings menu title cut off by scrollbar

### PR DESCRIPTION
**Pull request by JustinHoldstock**
Fixes scroll bar cutting off menu

screen shot 2018-01-22 at 4 58 54 pm

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @JustinHoldstock has signed the CLA. Thanks for the pull request!

File: src/lib/viewers/media/Settings.js [hunk 1]
```diff
@@ -292,11 +292,15 @@ class Settings extends EventEmitter {
      * @return {void}
      */
     setMenuContainerDimensions(menu) {
-        // NOTE: need to explicitly set the dimensions in order to get css transitions. width=auto doesn't work with css transitions
-        this.settingsEl.style.width = `${menu.offsetWidth + 18}px`;
+        const { children } = menu;
+
+        // If we have enough children to require scrolling, take into account scroll bar width.
+        const scrollPadding = children.length > 7 ? 32 : 18;
```
Comment (DanDeMicco) [inline src/lib/viewers/media/Settings.js]: Extract these magic numbers to constants

File: src/lib/viewers/media/Settings.js [hunk 2]
```diff
@@ -292,11 +292,15 @@ class Settings extends EventEmitter {
...
+        this.settingsEl.style.width = `${menu.offsetWidth + scrollPadding}px`;
+
         // height = n * $item-height + 2 * $padding (see Settings.scss) + 2 * border (see Settings.scss)
         // where n is the number of displayed items in the menu
-        const sumHeight = [].reduce.call(menu.children, (sum, child) => sum + child.offsetHeight, 0);
+        const sumHeight = [].reduce.call(children, (sum, child) => sum + child.offsetHeight, 0);
```
Comment (DanDeMicco) [inline src/lib/viewers/media/Settings.js]: I was taking a look at this before and was trying to figure out why it iterates over the children and sums up the offsetHeight vs taking the menu.offsetHeight. Seemed to work with the menu offsetHeight. Any ideas?

File: src/lib/viewers/media/Settings.js (hunk 1 above)
Comment (jeremypress) [reply src/lib/viewers/media/Settings.js]: Would looking at scrollHeight help? 

`The Element.scrollWidth read-only property is a measurement of the width of an element's content, including content not visible on the screen due to overflow.`

https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollWidth

File: src/lib/viewers/media/__tests__/Settings-test.js [hunk 3]
```diff
@@ -101,6 +101,47 @@ describe('lib/viewers/media/Settings', () => {
...
+            menuEl.appendChild(document.createElement('span'));
+            settings.setMenuContainerDimensions(menuEl);
+
+            expect(settings.settingsEl.style.width).to.equal('18px');
+        });
+
+        it('should add extra padding to settingsEl based on menu contents that require scroll bar', () => {
+            const menuEl = document.createElement('div');
+            // Greater than enough to add a scroll bar.
+            menuEl.appendChild(document.createElement('span'));
```
Comment (jeremypress) [inline src/lib/viewers/media/__tests__/Settings-test.js]: Add one element and set it's size with js?

File: src/lib/viewers/media/Settings.js (hunk 2 above)
Comment (JustinHoldstock) [reply src/lib/viewers/media/Settings.js]: I'm not too sure, but I can confirm that `menu.offsetHeight` is the way to go.

File: src/lib/viewers/media/__tests__/Settings-test.js (hunk 3 above)
Comment (JustinHoldstock) [reply src/lib/viewers/media/__tests__/Settings-test.js]: This will work especially since we're no longer counting child heights, and using offsetHeight

**Pull request merged**